app/src/main/java/com/android/brewr/ui/explore/Map.kt

```diff
@@ -80,3 +87,23 @@ private suspend fun getCurrentLocation(context: Context, onSuccess: (LatLng) ->
     onSuccess(LatLng(46.5197, 6.6323))
   }
 }
+
+@Composable
+private fun getMarkerIcon(shopName: String): BitmapDescriptor {
+  val context = LocalContext.current
+
+  // Helper function to load and resize the icon
+  fun loadAndResizeIcon(resourceId: Int): BitmapDescriptor {
+    val originalBitmap = BitmapFactory.decodeResource(context.resources, resourceId)
+    val resizedBitmap = Bitmap.createScaledBitmap(originalBitmap, 100, 100, false)
+    return BitmapDescriptorFactory.fromBitmap(resizedBitmap)
+  }
+
+  return when {
+    shopName.contains("Starbucks", ignoreCase = true) ->
```

Comment:
I thougt that we needed to extract manually for a certain location using the API. I was worried how are you going to deal with it, but this is actually a genius idea. You simply check if the coffeeshop name contains the certain location such as "McDonald's" or "Starbucks". Very simple and smart code !